src/WPF/ArcGISRuntime.WPF.Samples/Samples/Data/EditAndSyncFeatures/EditAndSyncFeatures.xaml.cs

```diff
@@ -0,0 +1,511 @@
+                    message += ": " + string.Join<string>("\n", m);
+                }
+
+                // Show the message
+                ShowStatusMessage(message);
+            }
+        }
+
+        private void HandleSyncStatusChange(SyncGeodatabaseJob job)
+        {
+            JobStatus status = job.Status;
+
+            // Tell the user about job completion
+            if (status == JobStatus.Succeeded)
+            {
+                ShowStatusMessage("Sync task completed");
+            }
+
+            // See if the job failed
+            if (status == JobStatus.Failed)
+            {
+                // Create a message to show the user
+                string message = "Sync geodatabase job failed";
+
+                // Show an error message (if there is one)
+                if (job.Error != null)
+                {
+                    message += ": " + job.Error.Message;
+                }
+                else
+                {
+                    // If no error, show messages from the job
+                    var m = from msg in job.Messages select msg.Message;
+                    message += ": " + string.Join<string>("\n", m);
+                }
+
+                // Show the message
+                ShowStatusMessage(message);
+            }
+        }
+
+        private void SyncGeodatabase()
+        {
+            // Return if not ready
+            if (_readyForEdits != EditState.Ready) { return; }
+
+            // Create parameters for the sync task
+            SyncGeodatabaseParameters parameters = new SyncGeodatabaseParameters()
+            {
+                GeodatabaseSyncDirection = SyncDirection.Bidirectional,
+                RollbackOnFailure = false
+            };
+
+            // Add each layer to the sync job
+            foreach (GeodatabaseFeatureTable table in _resultGdb.GeodatabaseFeatureTables)
+            {
+                // Get the ID for the layer
+                long id = table.ServiceLayerId;
+
+                // Create the SyncLayerOption
+                SyncLayerOption option = new SyncLayerOption(id);
+
+                // Add the option
+                parameters.LayerOptions.Add(option);
+            }
+
+            // Create job
+            SyncGeodatabaseJob job = _gdbSyncTask.SyncGeodatabase(parameters, _resultGdb);
+
+            // Subscribe to status updates
+            job.JobChanged += Job_JobChanged;
+
+            // Subscribe to progress updates
+            job.ProgressChanged += Job_ProgressChanged;
+
+            // Start the sync
+
+            job.Start();
+        }
+
+        private void Job_ProgressChanged(object sender, EventArgs e)
+        {
+            // Get the job object
+            SyncGeodatabaseJob job = sender as SyncGeodatabaseJob;
+
+            // Update the progress bar
+            UpdateProgressBar(job.Progress);
+        }
+
+        // Get the path to the tile package used for the basemap
```

Comment:
Perhaps some more comments to let the user know that this is plumbing code used by the Sample Viewer and not part of the ArcGIS Runtime .NET API itself.